Orange/base.py

```diff
@@ -173,8 +173,8 @@ def active_preprocessors(self):
                 self.preprocessors is not type(self).preprocessors):
             yield from type(self).preprocessors
 
-    def check_learner_adequacy(self, _):
```

Comment:
This is dangerous. You are changing a signature of a Learner base class method. That would need deprecation.

Perhaps make a new separate method that just returns the error, like `explain_inadequacy(domain)`. If it returns None, there is no error. :) Then `check_learner_adequacy()` may be just `return explain_inadequacy(..) == None`.

And if we are making new functions, I really dislike the current term, "adequate". I'd replace the current `check_learner_adequacy` with `fits` or `can_fit`: `learner.fits(domain)`, isn't it nicely readable? Maybe "can_fit" would promise too much - what if fitting crashed? :) Perhaps `is_compatible`...

The error message (or None) function might rather be something like `explain_incompatibility`.

@janezd, which names would you choose?

Reply:
`incompatility_reason` is fine.